scripts/aggregate-themes/theme-aggregator.js

```diff
@@ -27,133 +28,144 @@ class ThemeAggregator {
    * @returns {string|null} - The output path of the aggregated theme file. Null if not generated.
    */
   static aggregate(theme) {
-    let themeConfig;
+    let themeConfig = {};
     const defaultConfig = path.resolve(process.cwd(), CONFIG);
     if (fs.existsSync(defaultConfig)) {
       // eslint-disable-next-line global-require, import/no-dynamic-require
       themeConfig = require(defaultConfig);
-      const assets = ThemeAggregator.aggregateThemes({ ...themeConfig, ...theme && { theme } });
-      if (assets) {
-        return ThemeAggregator.writeJsFile(assets);
-      }
     }
+
+    const themeAssets = ThemeAggregator.aggregateThemes({
+      ...themeConfig,
+      ...theme && { theme },
+    });
+
+    if (themeAssets) {
+      return ThemeAggregator.writeJsFile(themeAssets);
+    }
+
     return null;
   }
 
   /**
-   * Aggregates theme assets.
-   * @param {string} theme - The theme to aggregate.
+   * Aggregates theme assets into a js file.
    * @param {Object} options - The aggregation options.
-   * @param {boolean} defaultFlag - Whether the theme to be generated is a root or scope theme. Guards against default theme and scope theme being equivalent.
-   * @returns {array} - An array of file names.
+   * @returns {array} - An array of aggregated theme files
    */
-  static aggregateTheme(themeName, options = {}, defaultFlag) {
-    const { theme, scoped = [] } = options;
-
-    if (!themeName) {
+  static aggregateThemes(options) {
+    if (!ThemeAggregator.validate(options)) {
       return null;
     }
 
-    const isRoot = themeName === theme && defaultFlag;
-    const isScoped = scoped.indexOf(themeName) > -1;
-    const file = isScoped && !isRoot ? SCOPED_THEME : ROOT_THEME;
-    const assets = ThemeAggregator.find(`**/themes/${themeName}/${file}`, options);
+    // This creates a generated-themes directory to house theme assets.
+    fs.ensureDir(OUTPUT_DIR, (err) => {
```

Comment:
It took me a bit to track this down, but when validating my webpack configuration with Jest, I was seeing this error:

![Screen Shot 2020-01-22 at 9 22 33 AM](https://user-images.githubusercontent.com/14099737/72907181-141a1b80-3cf9-11ea-9bd0-e4124a884fbe.png)

Could we update this to use Promise syntax to resolve this error? The change I tested looks like:
```js
fs.ensureDir(OUTPUT_DIR,).catch((err) => {
    Logger.warn(err, { LOG_CONTEXT });
});
```

Reply:
oops, good catch https://github.com/cerner/terra-toolkit/pull/376/commits/7a3892b2f4e5b5c2a5153b76c9d4f736c696596a

Reply:
much thanks!